compiler/qsc_hir/src/ty.rs

```diff
@@ -376,7 +376,7 @@ impl Display for FunctorSet {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         match self {
             Self::Value(value) => Display::fmt(value, f),
-            Self::Param(param) => Display::fmt(param, f),
+            Self::Param(param) => write!(f, "Param<{param}>"),
```

Comment:
nitpick: Maybe a slightly different name in the string here would help different these from type parameters? Looking at the tests where it shows up, initially I was confused why a type paramter would appear after a return type, ie `Unit is Param<0>` but then eventually realized it was a functor set. Perhaps `FuncSet<{param}>`?

That said, I don't feel strongly about it, so if you want to leave it as-is for consistency with the Rust enum name and/or the other params, I'm fine with it.

Reply:
Do you know why the ids for functor sets are lumped in with the ids for generics. Seems like a strange pairing to me.

Reply:
Discussed offline: because the functor sets and type parameters share the same sequencing for their ids, it is confusing to label them differently. Sticking with `Param<0>` for now.